pkg/queues/provider.go

```diff
@@ -0,0 +1,24 @@
+package queues
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Provider interface {
+	NewConsumer(queueName string) (Consumer, error)
+	NewPublisher(queueName string) (Publisher, error)
```

Comment:
Do the Consumer and Publisher constructors need to be in the interface?

Reply:
They rely on the provider since they are using variables that are part of the provider.

Reply:
Can you decouple those variables? Generally, constructors are functions.

```go
consumer, err := NewConsumer(log )
// handle err

provider.AddQueue(consumer)
```

Reply:
If the provider holds data that it must share it should expose it as a method.

url := provider.URL()
consumer, err := NewConsumer(log, url)

Reply:
I thought that provider is a factory for Consumer/Producer